tests/unittests/OperatorTest.cpp

```diff
@@ -69,6 +69,262 @@ static Tensor createTensorConditionallyQuantized(ElemKind T,
+
+  auto *save = F->createSave("save", cmpr);
+  auto *saveAlloc = bindings.allocate(save->getPlaceholder());
+
+  EE.compile(CompilationMode::Infer);
+  EE.run(bindings);
+
+  return saveAlloc->getHandle<bool>();
+}
+
+TEST_P(OperatorTest, less_int8) {
+  ENABLED_BACKENDS(Interpreter, CPU);
+
+  llvm::SmallVector<int8_t, 16> xValues = {
+      3, 2, 3, 6, 4, 5, 6, 3, 7, 8, 9, 2, 3, 5, 7, 1,
+
+      1, 2, 3, 6, 4, 5, 6, 3, 7, 8, 9, 2, 3, 5, 7, 1,
+
+      1, 2, 3, 6, 4, 5, 6, 3, 7, 8, 9, 2, 3, 5, 7, 1,
+
+      1, 2, 3, 6, 4, 5, 6, 3, 7, 8, 9, 2, 3, 5, 7, 1};
+
+  llvm::SmallVector<int8_t, 16> yValues = {
+      3, 4, 5, 7, 2, 5, 0, 6, 4, 2, 1, 8, 5, 9, 2, 6,
+
+      3, 4, 5, 7, 2, 1, 0, 6, 4, 2, 1, 8, 5, 9, 2, 6,
+
+      3, 4, 5, 7, 2, 1, 0, 6, 4, 2, 1, 8, 5, 9, 2, 6,
+
+      3, 4, 5, 7, 2, 1, 0, 6, 4, 2, 1, 8, 5, 9, 2, 6};
+
+  llvm::SmallVector<size_t, 4> xDims = {2, 2, 4, 4};
+  llvm::SmallVector<size_t, 4> yDims = {2, 2, 4, 4};
+
+  Handle<bool> saveH =
+      lessHelper<int8_t>(bindings_, mod_, F_, EE_, ElemKind::Int8QTy, xValues,
+                         yValues, xDims, yDims);
+
+  llvm::SmallVector<bool, 16> refResults = {
+      false, true,  true,  true, false, false, false, true,
+      false, false, false, true, true,  true,  false, true,
+
+      true,  true,  true,  true, false, false, false, true,
+      false, false, false, true, true,  true,  false, true,
+
+      true,  true,  true,  true, false, false, false, true,
+      false, false, false, true, true,  true,  false, true,
+
+      true,  true,  true,  true, false, false, false, true,
+      false, false, false, true, true,  true,  false, true,
+  };
+
+  int counter = 0;
+  for (size_t i = 0; i < saveH.dims()[0]; ++i) {
+    for (size_t j = 0; j < saveH.dims()[1]; ++j) {
+      for (size_t k = 0; k < saveH.dims()[2]; ++k) {
+        for (size_t f = 0; f < saveH.dims()[3]; ++f) {
+          EXPECT_FLOAT_EQ(refResults[counter++], saveH.at({i, j, k, f}));
+        }
+      }
+    }
+  }
+}
+
+TEST_P(OperatorTest, less_floatCases) {
+  ENABLED_BACKENDS(Interpreter, CPU);
+
+  llvm::SmallVector<float, 16> xValues = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f};
+
+  llvm::SmallVector<float, 16> yValues = {5.0f, 4.0f, 3.0f, 2.0f, 1.0f};
+
+  llvm::SmallVector<size_t, 4> xDims = {1, 1, 1, 5};
+  llvm::SmallVector<size_t, 4> yDims = {1, 1, 1, 5};
+
+  Handle<bool> saveH =
+      lessHelper<float>(bindings_, mod_, F_, EE_, ElemKind::FloatTy, xValues,
+                        yValues, xDims, yDims);
+
+  llvm::SmallVector<bool, 16> refResults = {true, true, false, false, false};
+
+  int counter = 0;
+  for (size_t i = 0; i < saveH.dims()[0]; ++i) {
+    for (size_t j = 0; j < saveH.dims()[1]; ++j) {
+      for (size_t k = 0; k < saveH.dims()[2]; ++k) {
+        for (size_t f = 0; f < saveH.dims()[3]; ++f) {
+          EXPECT_FLOAT_EQ(refResults[counter++], saveH.at({i, j, k, f}));
+        }
+      }
+    }
+  }
+}
+
+TEST_P(OperatorTest, less_float) {
```

Comment:
This and `less_int8` seem to be very similar tests. Can we combine them similar to how we do with e.g. `testBatchedReduceZeroDimResult` where we reuse the same data? Then we could also test `Int32ITy` and `Float16Ty` too without a lot of extra code.

Reply:
This way there is some code duplication on data creation side, but it allows us to re-use the same function to test different shapes. I guess you can just have one data set and then come up with shapes that will map to it, but I think it will be limiting.